include/FanFicArchive.py

```diff
@@ -0,0 +1,85 @@
+from bs4 import BeautifulSoup; from urllib import request; from re import match; import requests
+
+parser = "lxml"
+
+def fanfic_req(link): return BeautifulSoup(request.urlopen(link).read(), parser)
+def fanfic_check(id, name):
+    if isinstance(id, int): id = str(id)
+    else: assert isinstance(id, str), "Error for" + name + ": id not int or str"
+
+def ao3_rip(ao3_id): # downloads whole story in one go
+    fanfic_check(ao3_id, "ao3_rip")
+    text = fanfic_req('http://archiveofourown.org/works/'+ao3_id+'?view_adult=true&view_full_work=true')
+    tag_bar = text.find("dl", class_="stats").find_all("dd") # everything in the top bar
+    tag_list = {}
+    for i in tag_bar:
+        tag = [j.get_text() for j in i.find_all("a", class_="tag")]
+        if tag == []: tag = [i.get_text().lstrip('\n').lstrip(' ').rstrip(' ').rstrip('\n')]
+        tag_list[i.get("class")[0]] = tag
+    tag_list['author'] = [i.get_text() for i in text.find_all("a", rel="author")]
+    summary = text.find_all("div", class_="summary module", role="complementary") # just below top bar
+    chapter = {}
+    for i in text.find_all("div", class_="chapter", id=True): # separating each chapters
+        x = i.find("h3", class_="title").find("a").get("href").split("/")[-1]
```

Comment:
Should be using `urllib.parse` to do this rather than manually splitting the URL

Reply:
can't disagree, would be great if example is shown

Reply:
```
>>> import posixpath
>>> from urllib.parse import urlparse
>>> posixpath.split(urlparse("https://www.some-fanfic.net/posts/12345?no-js=true#heading3").path)[-1]
'12345'
```
This is the type of thing I mean. It may be a little overly cautious but it ensures the proper semantics for getting the last portion of the URL.